src/EntityFrameworkMock.NSubstitute/Helpers/ReflectionHelper.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EntityFrameworkMock.NSubstitute.Helpers
+{
+    public static class ReflectionHelper
+    {
+        /// <summary>
+        /// https://stackoverflow.com/questions/671968/retrieving-property-name-from-lambda-expression
+        /// </summary>
+        public static PropertyInfo GetPropertyInfo<TSource, TProperty>(
+            TSource source,
+            Expression<Func<TSource, TProperty>> propertyLambda)
+        {
+            Type type = typeof(TSource);
+
+            MemberExpression member = propertyLambda.Body as MemberExpression;
+            if (member == null)
+                throw new ArgumentException(string.Format(
```

Comment:
use `$""` instead `string.Format` (here and other places)

Reply:
Normally, I would agree with you - however, I lifted this directly from a SO post, and so I simply marked it as not being mine and left exactly as is.  Changing it feels a bit like quoting from Dickins, but changing one of the characters names :-)

Reply:
I could remove this entire class and use:

    dbSetSelector.Compile()(DbContextObject).Returns(mock.DbSet);

instead of

    var propertyInfo = Helpers.ReflectionHelper.GetPropertyInfo(DbContextObject, dbSetSelector);
    propertyInfo.GetValue(DbContextObject).Returns(mock.DbSet);

🙂